### PR TITLE
[CMAKE] Fix cmake build error related to #20669

### PR DESCRIPTION
**Pull request by LoSealL**
In this PR, we fixed cmake build error when generating python pip package.
- Add a patch python script for protobuf to generate necessary pb files, prevent importing error in `create_python_api.py`

- For MKL, we copy MKL shared binaries into python package, to make things easy to use. Also this could remove the MKL `IF-ELSE` in `tf_python.cmake` because we don't need to assign an external path for mkl any more.

### CODE REVIEW

File: tensorflow/contrib/cmake/external/protobuf.cmake [hunk 1]
```diff
@@ -16,7 +16,7 @@ include (ExternalProject)
 
 set(PROTOBUF_INCLUDE_DIRS ${CMAKE_CURRENT_BINARY_DIR}/protobuf/src/protobuf/src)
 set(PROTOBUF_URL https://github.com/google/protobuf.git)
-set(PROTOBUF_TAG v3.6.0)
+set(PROTOBUF_TAG ce044817c7ba0aea27c3fd8e496635d94d20a755)
```
Comment (gunan) [inline tensorflow/contrib/cmake/external/protobuf.cmake]: Please use the tag v3.6.0.1 instead

**Review by gunan — CHANGES_REQUESTED** [review]

**Comment by gunan** [comment]
I have one comment, but adding @mrry and @annarev for a more thorough review.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @gunan, @mrry, @gunan, @annarev: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

File: tensorflow/contrib/cmake/patches/protobuf/build_pb.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import sys
+import os
+import subprocess
+from pathlib import Path
+
+
+def generate_proto(source, require = True, protoc = None):
```
Comment (annarev) [inline tensorflow/contrib/cmake/patches/protobuf/build_pb.py]: Can you add descriptions for arguments, for e.g. I can't tell what "require" does just from reading the signature?
Even better if you can write add the comment in the following format:
"""One line description.

Args:
  arg name: description for arg
"""

Also, can you change protoc to an argument without default value? Seems like protoc is required. Having a default value makes it seems like it is ok to call 'generate_proto' without 'protoc'.

**Review by annarev — COMMENTED** [review]
Sorry for a long delay. I somehow missed this PR.

How is this different from protobufs built here:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/cmake/tf_python.cmake#L82

Why does it need this extra step?

File: tensorflow/contrib/cmake/patches/protobuf/build_pb.py (hunk 2 above)
Comment (LoSealL) [reply tensorflow/contrib/cmake/patches/protobuf/build_pb.py]: Ok, but actually this is copied from `setup.py` file in protobuf, and I wonder why this failure is ignored for so long? 
Maybe It's better to require `protobuf` to be installed before building the tensorflow python packet ?

**Comment by LoSealL** [comment]
After the merge of PR #21201, there are some conflicts in generating `tf_python_api` and `estimator_python_api`, I'm resolving this.

**Comment by LoSealL** [comment]
Since r1.10 and later version will remove cmake support, close this PR

**Pull request closed without merging**